src/lib.rs

```diff
@@ -1277,6 +1277,44 @@ impl Error {
         self.depth
     }
 
+    /// Inspect the underlying [`io::Error`] if there is one.
+    /// If the underlying error isn't an [`io::Error`], [`None`] is returned.
+    ///
+    /// ```rust,no-run
+    /// use walkdir::{WalkDir};
+    /// use std::io;
+    /// use std::path::Path;
+    ///
+    /// let mut it = WalkDir::new("foo").into_iter();
+    /// loop {
+    ///     let entry = match it.next() {
+    ///         None => break,
+    ///         Some(Ok(entry)) => entry,
+    ///         Some(Err(err)) => {
+    ///             println!("failed to access entry {}", err.path().unwrap_or(Path::new("")).display());
+    ///             if let Some(inner) = err.io_error() {
+    ///                 match inner.kind() {
+    ///                     io::ErrorKind::InvalidData => println!("entry contains invalid data: {}", inner),
+    ///                     io::ErrorKind::PermissionDenied => println!("Missing permission to read entry: {}", inner),
+    ///                     _ => println!("Unexpected error occurred: {}", inner),
+    ///                 }
+    ///             }
+    ///             continue
+    ///         }
+    ///     };
+    ///     println!("{}", entry.path().display());
```

Comment:
I feel like the `let entry = ...` here is unnecessary. Maybe just move the `println!` into the match expression arm?